src/cascadia/TerminalSettingsEditor/MainPage.cpp

```diff
@@ -0,0 +1,305 @@
+                }
+            }
+            else if (const auto profile = clickedItemContainer.Tag().try_as<Model::Profile>())
+            {
+                // Navigate to a page with the given profile
+                contentFrame().Navigate(xaml_typename<Editor::Profiles>(), winrt::make<ProfilePageNavigationState>(profile, _settingsClone.GlobalSettings().ColorSchemes(), *this));
+            }
+        }
+    }
+
+    void MainPage::_Navigate(hstring clickedItemTag)
+    {
+        if (clickedItemTag == launchTag)
+        {
+            contentFrame().Navigate(xaml_typename<Editor::Launch>(), winrt::make<LaunchPageNavigationState>(_settingsClone));
+        }
+        else if (clickedItemTag == interactionTag)
+        {
+            contentFrame().Navigate(xaml_typename<Editor::Interaction>(), winrt::make<InteractionPageNavigationState>(_settingsClone.GlobalSettings()));
+        }
+        else if (clickedItemTag == renderingTag)
+        {
+            contentFrame().Navigate(xaml_typename<Editor::Rendering>(), winrt::make<RenderingPageNavigationState>(_settingsClone.GlobalSettings()));
+        }
+        else if (clickedItemTag == globalProfileTag)
+        {
+            contentFrame().Navigate(xaml_typename<Editor::Profiles>(), winrt::make<ProfilePageNavigationState>(_settingsClone.ProfileDefaults(), _settingsClone.GlobalSettings().ColorSchemes(), *this));
+        }
+        else if (clickedItemTag == colorSchemesTag)
+        {
+            contentFrame().Navigate(xaml_typename<Editor::ColorSchemes>(), winrt::make<ColorSchemesPageNavigationState>(_settingsClone.GlobalSettings()));
+        }
+        else if (clickedItemTag == globalAppearanceTag)
+        {
+            contentFrame().Navigate(xaml_typename<Editor::GlobalAppearance>(), winrt::make<GlobalAppearancePageNavigationState>(_settingsClone.GlobalSettings()));
+        }
+    }
+
+    void MainPage::OpenJsonTapped(IInspectable const& /*sender*/, Windows::UI::Xaml::Input::TappedRoutedEventArgs const& /*args*/)
+    {
+        const CoreWindow window = CoreWindow::GetForCurrentThread();
+        const auto rAltState = window.GetKeyState(VirtualKey::RightMenu);
+        const auto lAltState = window.GetKeyState(VirtualKey::LeftMenu);
+        const bool altPressed = WI_IsFlagSet(lAltState, CoreVirtualKeyStates::Down) ||
+                                WI_IsFlagSet(rAltState, CoreVirtualKeyStates::Down);
+
+        const auto target = altPressed ? SettingsTarget::DefaultsFile : SettingsTarget::SettingsFile;
+        _OpenJsonHandlers(nullptr, target);
+    }
+
+    void MainPage::OpenJsonKeyDown(IInspectable const& /*sender*/, Windows::UI::Xaml::Input::KeyRoutedEventArgs const& args)
+    {
+        if (args.Key() == VirtualKey::Enter || args.Key() == VirtualKey::Space)
+        {
+            const auto target = args.KeyStatus().IsMenuKeyDown ? SettingsTarget::DefaultsFile : SettingsTarget::SettingsFile;
+            _OpenJsonHandlers(nullptr, target);
+        }
+    }

```

Comment:
Wait don't buttons just handle a Space or Enter themselves? I didn't think you had to do this manually

Reply:
_Surprisingly_, this is also necessary. Removing the KeyDown handler _completely_ removed functionality for invoking it with the keyboard. Adding a comment in the .xaml file.

Reply:
w a t. Okay, thanks for the follow up.